### PR TITLE
fix: only include necessary files into npm

### LINKED ISSUE #238: All files are included into npm

### PR DESCRIPTION
**Pull request by JounQin**
close #238

cc @ljharb

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,11 @@
 		"url": "git://github.com/browserify/resolve.git"
 	},
 	"main": "index.js",
+	"files": [
+		"/lib",
+		"/index.js",
+		"/index.mjs"
+	],
```
Comment (ljharb) [inline package.json:14]: "files" is highly dangerous and should never be used; only `.npmignore` should be used to control what files are published.

**Review by ljharb — CHANGES_REQUESTED** [review]
This would be a breaking change.

This is quite intentional. Those files must be included.

> Tests should be published with every package, so that `npm explore foo && npm install && npm test` always works.
>
> Installed package size is irrelevant; if you never require the test files, they'll never affect the runtime of your application.

Duplicate of #235. Duplicate of #58. Duplicate of #44. See https://github.com/browserify/resolve/pull/105#issuecomment-266653976.

File: package.json (hunk 1 above)
Comment (JounQin) [reply package.json:14]: Highly disagree, but you're the boss here. 🧐

Comment (ljharb) [reply package.json:14]: "a lot of packages use it" isn't an argument for it being safe.

The worst possible damage caused by accidental omission from "npmignore" is "the package is slightly larger". The worst possible damage caused by accidental omission from "files" is "every consumer breaks".

**Pull request closed without merging**